src/library_async.js

```diff
@@ -660,15 +706,26 @@ mergeInto(LibraryManager.library, {
       Asyncify.dataInfo[ptr] = {
         bottomOfCallStack: bottomOfCallStack
       };
+#if ASYNCIFY_DEBUG >= 2
+      err('ASYNCIFY: dataInfo after allocateData('+ptr+'):', Asyncify.dataInfo);
+#endif
       return ptr;
     },
 
     freeData: function(ptr) {
+#if ASYNCIFY_DEBUG >= 2
+      if (Asyncify.dataInfo[ptr] === undefined) {
+        err('ASYNCIFY: called freeData on unknown pointer', ptr, new Error().stack);
+      }
+#endif
       _free(ptr);
-      Asyncify.dataInfo[ptr] = null;
```

Comment:
There is no particularly strong reason. My original motivation was the theoretical possibility that in case you have a lot of short-lived fibers and frequent swaps, you might end up with some dead keys in the mapping that might never be reused, wasting memory. That may happen if a random `malloc()` inside the application just happens to return an address that's `null`'d in the mapping.

This is probably not a big deal, and I don't know about the performance implications. So if you think it should be changed into `null` I have no other objections.

Reply:
I see. That sounds reasonable, let's leave it like it is.